src/types.ts

```diff
@@ -75,35 +75,53 @@ type SectionType = 'autocomplete' | 'recommendations' | 'custom';
 
 export type SectionConfiguration = {
   type?: SectionType;
-  identifier: string;
   displayName?: string;
   numResults?: number;
   // This property will only take effect when using the component and not the hook
   displaySearchTermHighlights?: boolean;
   ref?: React.RefObject<HTMLElement>;
 };
 
-export interface AutocompleteSection extends SectionConfiguration {
+export interface AutocompleteSectionConfiguration extends SectionConfiguration {
   type?: 'autocomplete';
-  data: Item[];
+  indexSection: string;
+  /** @deprecated use indexSection field instead */
+  identifier?: string;
 }
 
-export interface RecommendationsSection extends SectionConfiguration {
+export interface RecommendationsSectionConfiguration extends SectionConfiguration {
   type: 'recommendations';
-  data: Item[];
+  indexSection: string;
+  podId: string;
   itemIds?: string[];
-  section?: string;
   term?: string;
+  /** @deprecated use podId field instead */
+  identifier?: string;
 }
 
-export interface CustomSection extends SectionConfiguration {
+export interface CustomSectionConfiguration extends SectionConfiguration {
   type: 'custom';
+  displayName: string;
+}
+
+export interface AutocompleteSection extends AutocompleteSectionConfiguration {
+  data: Item[];
+}
+
+export interface RecommendationsSection extends RecommendationsSectionConfiguration {
+  data: Item[];
+}
+
+export interface CustomSection extends CustomSectionConfiguration {
```

Comment:
Having section vs sectionConfiguration make sense for the other sections that gets data added to it later. But for CustomSectionConfiguration `data` is already part of the configuration. What do you think about adding `data: Item[];` to `CustomSectionConfiguration` and remove `CustomSection` then replace occurrences of `CustomSection` -> `CustomSectionConfiguration`

Reply:
This makes sense. I was trying to keep the change non-breaking. Considering we don't have anyone using a custom section at the moment, I guess we can make a breaking change